src/Instrumentation/Psr15/src/Psr15Instrumentation.php

```diff
@@ -0,0 +1,129 @@
+                if (!$request instanceof ServerRequestInterface) {
+                    //`pre` hook runs before params are checked, so this is theoretically possible. activate a non-recording
+                    //span so that the post handler has something to close
+                    $root = new NonRecordingSpan();
+                    $root->activate();
+                    Context::storage()->attach($root->storeInContext(Context::getCurrent()));
+                    return;
+                }
+                $root = $request->getAttribute(SpanInterface::class);
+                $builder = $instrumentation->tracer()->spanBuilder(
+                        $root
+                        ? sprintf('%s::%s', $class, $function)
+                        : sprintf('HTTP %s', $request?->getMethod() ?? 'unknown')
+                    )
+                    ->setSpanKind(SpanKind::KIND_SERVER)
+                    ->setAttribute('code.function', $function)
+                    ->setAttribute('code.namespace', $class)
+                    ->setAttribute('code.filepath', $filename)
+                    ->setAttribute('code.lineno', $lineno);
+                $parent = Context::getCurrent();
+                if (!$root) {
+                    //create http root span
+                    $parent = Globals::propagator()->extract($request->getHeaders());
+                    $span = $builder
+                        ->setParent($parent)
+                        ->setAttribute(TraceAttributes::HTTP_URL, $request?->getUri()->__toString())
+                        ->setAttribute(TraceAttributes::HTTP_METHOD, $request?->getMethod())
+                        ->setAttribute(TraceAttributes::HTTP_REQUEST_CONTENT_LENGTH, $request?->getHeaderLine('Content-Length'))
+                        ->setAttribute(TraceAttributes::HTTP_SCHEME, $request?->getUri()?->getScheme())
+                        ->startSpan();
+                    $request = $request->withAttribute(SpanInterface::class, $span);
+                } else {
+                    $span = $builder->startSpan();
+                }
+                Context::storage()->attach($span->storeInContext($parent));
+
+                return [$request];
+            },
+            post: static function (RequestHandlerInterface $handler, array $params, ?ResponseInterface $response, ?Throwable $exception) {
+                $scope = Context::storage()->scope();
+                if (!$scope) {
+                    return;
+                }
+                $scope->detach();
+                $span = Span::fromContext($scope->context());
+                if ($exception) {
+                    $span->recordException($exception, [TraceAttributes::EXCEPTION_ESCAPED => true]);
+                    $span->setStatus(StatusCode::STATUS_ERROR, $exception->getMessage());
+                }
+                if ($response?->getStatusCode() >= 500) {
```

Comment:
Should capture some additional [http attributes](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#common-attributes) such as `http.status_code`.

Reply:
done, added a few more attributes.